src/controllers/auth.controller.js

```diff
@@ -0,0 +1,268 @@
+const bcrypt = require('bcrypt');
+const uuid = require('uuid');
+
+require('dotenv').config();
+
+const emailService = require('../services/email.service');
+const userService = require('../services/user.service');
+const jwtService = require('../services/jwt.service');
+const tokenService = require('../services/token.service');
+const { ApiError } = require('../utils/api.error');
+const { User } = require('../models/user.model');
+
+const register = async (req, res) => {
+  const { name, email, password } = req.body;
+  const activationToken = uuid.v4();
+
+  const { refreshToken } = req.cookies;
+  const userData = await jwtService.validateRefreshToken(refreshToken);
+
+  if (userData) {
+    throw ApiError.badRequest('Only for not authorized');
+  }
+
+  if (!name || !email || !password) {
+    throw ApiError.badRequest('Not all credentials are provided!');
+  }
+
+  const user = await userService.getByEmail(email);
+
+  if (user) {
+    throw ApiError.badRequest('The user with this email already exists!');
+  }
+
+  const hashedPassword = await bcrypt.hash(
+    'password',
+    +process.env.HASH_ROUNDS,
```

Comment:
The password hashing is using a hardcoded string 'password' instead of the actual user-provided password. This is a critical issue as it means all users will have the same password hash. You should replace `'password'` with the `password` variable from the request body.